### PR TITLE
Add Xdt module to FAKE 5

### PR DESCRIPTION
**Pull request by panesofglass**
### Description

Migrate `Fake.XDTHelper` to FAKE 5 module `Fake.DotNet.Xdt`.

## TODO

Feel free to open the PR and ask for help

- [x] New (API-)documentation for new features exist (Note: API-docs are enough, additional docs are in `help/markdown`)
- [x] unit or integration test exists (or short reasoning why it doesn't make sense)
  
  > Note: Consider using the `CreateProcess` API which can be tested more easily, see https://github.com/fsharp/FAKE/pull/2131/files#diff-4fb4a77e110fbbe8210205dfe022389b for an example (the changes in the `DotNet.Testing.NUnit` module)
  
- [x] boy scout rule: "leave the code behind in a better state than you found it" (fix warnings, obsolete members or code-style in the places you worked in)     
- [x] (if new module) the module has been linked from the "Modules" menu, edit `help/templates/template.cshtml`, linking to the API-reference is fine.
- [x] (if new module) the module is in the correct namespace
- [x] (if new module) the module is added to Fake.sln (`dotnet sln Fake.sln add src/app/Fake.*/Fake.*.fsproj`)
- [x] Fake 5 [API guideline](https://fake.build/contributing.html#API-Design) is honored

### CODE REVIEW

**Comment by panesofglass** [comment]
@matthid looks like only CircleCI failed this time.

**Comment by matthid** [comment]
```
[17:19:17 ERR] Fake.DotNet.Xdt.Tests/when transforming files with config name errored in 00:00:00.2530000 <Expecto>
System.IO.IOException: The process cannot access the file '/build/src/test/Fake.Core.UnitTests/TestFiles/Fake.DotNet.Xdt.Files/web.config' because it is being used by another process.
   at System.IO.FileStream.Init(FileMode mode, FileShare share)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite)
   at System.IO.File.Copy(String sourceFileName, String destFileName, Boolean overwrite)
   at Fake.DotNet.XdtTests.tests@63-38.Invoke(Unit unitVar)
   at Expecto.Impl.execTestAsync@924-1.Invoke(Unit unitVar) in /Users/h/dev/logibit/expecto/Expecto/Expecto.fs:line 928
   at Microsoft.FSharp.Control.AsyncBuilderImpl.callA@522.Invoke(AsyncParams`1 args)
```

looks related to this PR.

**Comment by panesofglass** [comment]
@matthid parallel tests, perhaps?

**Comment by matthid** [comment]
Yes by default stuff runs parallel and I feel like unit tests should work when run in parallel ;). If for some strong reason they need to run sequential there is a sequentialTests function in expecto.

**Comment by panesofglass** [comment]
Thanks, @matthid. Looks like that fixed it.

File: src/test/Fake.Core.UnitTests/Fake.DotNet.Xdt.fs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    let copy source dest =
+        require source
+        File.Copy(fileName source, fileName dest, true)
+
+    let delete file =
+        File.Delete(fileName file)
+
+[<Tests>]
+let tests =
+    testSequenced <| testList "Fake.DotNet.Xdt.Tests" [
```
Comment (matthid) [inline src/test/Fake.Core.UnitTests/Fake.DotNet.Xdt.fs]: Is it possible to just copy the test-files or create unique names to remove the need to have them running sequentially?

Comment (panesofglass) [reply src/test/Fake.Core.UnitTests/Fake.DotNet.Xdt.fs]: I'm not entirely sure what you mean. Can you give an example? I copied the tests that were provided for the old `XDTHelper` and converted them to use Expecto.

File: src/app/Fake.DotNet.Xdt/Xdt.fs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    else
+        Trace.traceImportantfn "No %s config file found for '%s'. Skipping." configName xmlFile
+
+/// Modifies XML files in place using an XDT file named by inserting a .configName in between each filename and .extension.
+let transformFilesWithConfigName (configName:string) (files:Fake.IO.IGlobbingPattern) = 
+    Seq.iter (transformFileWithConfigName configName) files
+
+/// Reads XML file (typically a config file), makes changes according to XDT transform syntax, saves result.
+[<System.Obsolete("Please use transformFile instead.")>]
+let TransformFile (inXmlFile:string) (transformFileName:string) (outXmlFile:string) =
```
Comment (matthid) [inline src/app/Fake.DotNet.Xdt/Xdt.fs]: I think we could remove those

File: src/test/Fake.Core.UnitTests/Fake.DotNet.Xdt.fs (hunk 1 above)
Comment (panesofglass) [reply src/test/Fake.Core.UnitTests/Fake.DotNet.Xdt.fs]: One challenge for two of the functions is that they modify a file in-place, so I'm not sure how to resolve that without adding another dependency like a virtual file system.

Comment (panesofglass) [reply src/test/Fake.Core.UnitTests/Fake.DotNet.Xdt.fs]: I'm happy to make the change, but I'm not sure what exactly to do here. Do you have a specific suggestion?

**Comment by panesofglass** [comment]
@matthid, does my last commit address your concerns?

**Comment by panesofglass** [comment]
@mattid, would you mind re-running the AppVeyor build? The failure looks like a build server error.

File: src/app/Fake.DotNet.Xdt/Xdt.fs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+open System
+open System.IO
+open System.Xml
+open DotNet.Xdt
+open Fake.Core
+
+let private xdtTag = KnownTags.Task "XDT"
+
+/// Integrates XDT logging into FAKE logging.
+type FakeXmlTransformationLogger() =
```
Comment (matthid) [inline src/app/Fake.DotNet.Xdt/Xdt.fs]: Can/should we make that internal?

**Comment by panesofglass** [comment]
@matthid any chance you can merge this? It's blocking us from using FAKE 5 on our web projects.

**Comment by matthid** [comment]
Yes sorry, I just wanted to make `FakeXmlTransformationLogger` internal and take a look at the tests but I didn't have the opportunity yet.

**Comment by panesofglass** [comment]
@matthid, so long as change to making FakeXmlTransformationLogger internal doesn't break the build, I think this should be good. Thanks for your feedback!

**Comment by matthid** [comment]
@panesofglass please see/review my changes in https://github.com/fsharp/FAKE/pull/2221/commits/570e10b83b57b3a8f39475af23be643632c85cf7
Will release soonish.

**Comment by matthid** [comment]
Again thanks! Sorry for the delay.

**Pull request merged**